edi_oca/models/edi_configuration_mixin.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2024 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from odoo import api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class EDIConfigurationMixin(models.AbstractModel):
```

Comment:
I don't understand why we need this mixin. Is not in the specs... :/
Looks like a not required copy/past from `*_auto` module.

I think there's a big misunderstanding on WHO is supposed to trigger the specific action.
I wanted to simplify what I've done on `*_auto` module getting rid of all this automation and leave the control to the specific module and to the configuration that you can define.

Concept:
1. the trigger is given by the specific module -> `on_button_confirm_purchase_order` must be an event triggered by the module `edi_purchase_oca` when `button_confirm` is called (which is already the case!)
2. the condition to skip or exec a conf is defined on the conf itself -> that's why I described 2 methods (before_do and do)